packages/react/src/jsx/jsx-local-namespace.ts

```diff
@@ -68,7 +68,14 @@ export namespace CompiledJSX {
   export type IntrinsicAttributes = ReactJSXIntrinsicAttributes;
   export type IntrinsicClassAttributes<T> = ReactJSXIntrinsicClassAttributes<T>;
   export type IntrinsicElements = {
-    [K in keyof ReactJSXIntrinsicElements]: ReactJSXIntrinsicElements[K] & {
+    [K in keyof ReactJSXIntrinsicElements]: Omit<ReactJSXIntrinsicElements[K], 'className'> & {
+      // We override class name so we can pass xcss prop to it. We opt to do this instead of
+      // Making the output of cssMap() a string intersection so we can also have an inline object
+      // be declared.
+      /**
+       * The class name prop now can be given the output of xcss prop from `@compiled/react`.
+       */
+      className?: string | Record<string, any> | null | false;
```

Comment:
This is needed to enable passing xcss prop to className:

```jsx
function Component({ xcss }: { xcss?: XCSSProp<'color', never> }) {
  return <div className={xcss} />;
}
```

Else it is a type error. There was an alternate solution which was enabling xcss prop / css map types be intersected with a string `& string` but that would then not work with inline objects:

```jsx
<Component xcss={{ color: 'red' }} />
```

Since we own both ends (jsx pragma + compiled APIs) I opted for this solution to get it all working.